src/deepsparse/transformers/pipelines/nli_text_classification.py

```diff
@@ -218,16 +219,28 @@ def nli_engine_forward(
         pass
     :return: result of forward pass to Pipeline engine
     """
-    engine_inputs = numpy.array(engine_inputs)
 
-    engine_outputs = []
-    for batch_origin_index in range(0, engine_inputs.shape[1], pipeline._batch_size):
+    def _engine_forward(batch_origin_index: int, engine_outputs: List[numpy.ndarray]):
         labelwise_inputs = engine_inputs[
             :, batch_origin_index : batch_origin_index + pipeline._batch_size, :
         ]
         labelwise_inputs = [labelwise_input for labelwise_input in labelwise_inputs]
         engine_output = pipeline.engine(labelwise_inputs)  # TODO: Parallelize
         engine_outputs += engine_output
 
+    engine_inputs = numpy.array(engine_inputs)
+
+    engine_outputs = []
+    threads = []
+    for batch_origin_index in range(0, engine_inputs.shape[1], pipeline._batch_size):
+        thread = Thread(
+            target=_engine_forward, args=(batch_origin_index, engine_outputs)
+        )
```

Comment:
these Threads aren't guaranteed to execute sequentially obviously - is it important for the `engine_outputs += engine_output` appending to be ordered?

Reply:
Thanks for the catch!